File: backend/src/main/java/com/waterball/course/entity/Journey.java
package com.waterball.course.entity;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;
import java.util.UUID;

@Entity
@Table(name = "journeys")
@Getter @Setter
@NoArgsConstructor
public class Journey {
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @Column(nullable = false)
    private String title;

    private String description;

    @Column(name = "thumbnail_url")
    private String thumbnailUrl;

    @Column(name = "is_published")
    private Boolean isPublished = false;

    @Column(nullable = false, precision = 10, scale = 2)
    private BigDecimal price = BigDecimal.ZERO;

    @Column(name = "created_at")
    private LocalDateTime createdAt;

    @Column(name = "updated_at")
    private LocalDateTime updatedAt;

    @OneToMany(mappedBy = "journey", cascade = CascadeType.ALL, orphanRemoval = true)
    @OrderBy("sortOrder ASC")
    private List<Chapter> chapters = new ArrayList<>();

    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
        updatedAt = LocalDateTime.now();
    }

    @PreUpdate
    protected void onUpdate() {
        updatedAt = LocalDateTime.now();
    }
}

File: frontend/src/types/journey.ts
import { ChapterWithLessons } from './chapter';

export interface Journey {
  id: string;
  title: string;
  description: string | null;
  thumbnailUrl: string | null;
  chapterCount: number;
  lessonCount: number;
  totalDurationSeconds: number;
  price: number;
  currency: string;
  originalPrice?: number;
  discountPercentage?: number;
}

export interface JourneyDetail extends Journey {
  chapters: ChapterWithLessons[];
  isPurchased: boolean;
}
